### PR TITLE
Allow multiple credentials for Cardano testnets

### PR DESCRIPTION
**Pull request by mrBliss**
Allow multiple Shelley-based credentials for Cardano, but only for testnets, not for mainnnet.

### CODE REVIEW

File: ouroboros-consensus/src/Ouroboros/Consensus/Util/OptNP.hs
Comment (mrBliss) [inline ouroboros-consensus/src/Ouroboros/Consensus/Util/OptNP.hs:230]: Too ad hoc, move it to `Cardano/Node.hs`, make it more specific and document what it is doing.

**Review by edsko — APPROVED** [review]

**Comment by mrBliss** [comment]
bors merge

**Comment by mrBliss** [comment]
@nfrisby Edsko and I thought we should review this with you, but @deepfire is blocked on this.

**Pull request merged**
